File: solutions/242ValidAnagram/main.go
/*
Given two strings s and t, return true if t is an anagram
of s, and false otherwise.
*/

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	counts := make(map[rune]int, len(s))
	for _, char := range []rune(s) {
		counts[char]++
	}
	for _, char := range []rune(t) {
		if counts[char] == 0 {
			return false
		}
		counts[char]--
	}
	return true
}

File: solutions/242ValidAnagram/main.py
'''
Given two strings s and t, return true if t is an anagram
of s, and false otherwise.
'''

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        for letter in set(s):
            if s.count(letter) != t.count(letter):
                return False
        return True
